libbeat/common/cli/cli.go

```diff
@@ -0,0 +1,20 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+// RunWith wrap cli function with an error handler instead of having the code exit early.
+func RunWith(
+	fn func(cmd *cobra.Command, args []string) error,
+) func(cmd *cobra.Command, args []string) {
+	return func(cmd *cobra.Command, args []string) {
+		if err := fn(cmd, args); err != nil {
```

Comment:
I wonder if we want to handle/capture panic here as well?

Reply:
I've took some time to think about, even if it's not necessary because the runtime would take care of the panic and do the right thing. Logic wise it will be cleaner to manually manage that in the `RunWith` and manually log what happened.